server/src/analyser.ts

```diff
@@ -332,6 +457,179 @@ export default class Analyzer {
     return locations
   }
 
+  /**
+   * A more scope-aware version of findOccurrences.
+   */
+  public findOccurrencesWithin({
```

Comment:
I didn't look to carefully here, but it might be we can reuse some functionality from `utils/declarations`.